controllers/designate_controller.go

```diff
@@ -1059,9 +1062,59 @@ func (r *DesignateReconciler) generateServiceConfigMaps(
 	// - %-config configmap holding minimal designate config required to get the service up, user can add additional files to be added to the service
 	// - parameters which has passwords gets added from the ospSecret via the init container
 	//
+	Log := r.GetLogger(ctx)
 
 	cmLabels := labels.GetLabels(instance, labels.GetGroupLabel(designate.ServiceName), map[string]string{})
+	replicas := int(*instance.Spec.DesignateBackendbind9.Replicas)
+	secretName := designate.DesignateBindKeySecret
+
+	// Try to get the existing secret first, if not found, create a new one
+	secret := &corev1.Secret{}
+	err := h.GetClient().Get(ctx, client.ObjectKey{
+		Name:      secretName,
+		Namespace: instance.Namespace,
+	}, secret)
+
+	if err != nil {
+		if !k8s_errors.IsNotFound(err) {
+			Log.Info("Error when trying to get the secret")
+			return err
+		}
+		keysMap := map[string][]byte{}
+		secret = &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      secretName,
+				Namespace: instance.Namespace,
+				Labels:    cmLabels,
+			},
+			Type: corev1.SecretTypeOpaque,
+			Data: keysMap,
+		}
+		if err := h.GetClient().Create(ctx, secret); err != nil {
+			if !k8s_errors.IsAlreadyExists(err) {
+				return err
+			}
+		}
+	}
+	numOfKeys := len(secret.Data)
+	keysDelta := int(math.Abs(float64(numOfKeys) - float64(replicas)))
```

Comment:
now that keysDelta is only used in the numOfKeys > replicas case, you don't need all those conversions
you can directly pass (numOfKeys - replicas)